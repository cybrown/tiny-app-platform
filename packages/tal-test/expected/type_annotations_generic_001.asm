main()
  entry:
    FunctionRef       name: myGenericFunc0_0
    DeclareLocal      name: myGenericFunc0, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    FunctionRef       name: myGenericFunc1_1
    DeclareLocal      name: myGenericFunc1, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    FunctionRef       name: myGenericFunc2_2
    DeclareLocal      name: myGenericFunc2, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    FunctionRef       name: myGenericFunc3_3
    DeclareLocal      name: myGenericFunc3, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    FunctionRef       name: myGenericFunc4_4
    DeclareLocal      name: myGenericFunc4, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    FunctionRef       name: myGenericFunc5_5
    DeclareLocal      name: myGenericFunc5, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    FunctionRef       name: myGenericFunc6_6
    DeclareLocal      name: myGenericFunc6, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    FunctionRef       name: myGenericFunc7_7
    DeclareLocal      name: myGenericFunc7, mutable: false, hasInitialValue: true
myGenericFunc0_0(param)
  entry:
    Literal           1
    MakeArrayForBlock count: 1
myGenericFunc1_1(param)
  entry:
    Literal           2
    MakeArrayForBlock count: 1
myGenericFunc2_2(param)
  entry:
    Literal           3
    MakeArrayForBlock count: 1
myGenericFunc3_3(param)
  entry:
    Literal           null
    MakeArrayForBlock count: 1
myGenericFunc4_4(param)
  entry:
    Local             param
    MakeArrayForBlock count: 1
myGenericFunc5_5(param)
  entry:
    Literal           null
    MakeArrayForBlock count: 1
myGenericFunc6_6(param)
  entry:
    Local             param
    MakeArrayForBlock count: 1
myGenericFunc7_7(param)
  entry:
    Local             param
    MakeArrayForBlock count: 1
