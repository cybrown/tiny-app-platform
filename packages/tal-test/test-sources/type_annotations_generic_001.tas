// Is generic, but do not use type param
fun myGenericFunc0<T>(param: number): number {
    1
}

// Is generic with many generic parameters, but do not use type param
fun myGenericFunc1<T, U, V>(param: number): number {
    2
}

// Use type param in parameter
fun myGenericFunc2<T>(param: T): number {
    3
}

// Use type param in return type
fun myGenericFunc3<T>(param: number): T {
    null
}

// Use type param in parameter and return type
fun myGenericFunc4<T>(param: T): T {
    null
}
