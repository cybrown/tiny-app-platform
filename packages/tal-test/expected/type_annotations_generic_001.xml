<program>
  <comment>
    <declarelocal>
      <function>
        <block>
          <literal />
        </block>
      </function>
    </declarelocal>
  </comment>
  <comment>
    <declarelocal>
      <function>
        <block>
          <literal />
        </block>
      </function>
    </declarelocal>
  </comment>
  <comment>
    <declarelocal>
      <function>
        <block>
          <literal />
        </block>
      </function>
    </declarelocal>
  </comment>
  <comment>
    <declarelocal>
      <function>
        <block>
          <literal />
        </block>
      </function>
    </declarelocal>
  </comment>
  <comment>
    <declarelocal>
      <function>
        <block>
          <local />
        </block>
      </function>
    </declarelocal>
  </comment>
  <comment>
    <declarelocal>
      <function>
        <block>
          <literal />
        </block>
      </function>
    </declarelocal>
  </comment>
  <comment>
    <declarelocal>
      <function>
        <block>
          <local />
        </block>
      </function>
    </declarelocal>
  </comment>
  <comment>
    <declarelocal>
      <function>
        <block>
          <local />
        </block>
      </function>
    </declarelocal>
  </comment>
</program>
